### PR TITLE
chore(deps): bump sarama to v1.41.2

### PR DESCRIPTION
**Pull request by dnwe**
## Proposed Changes

chore(deps): bump sarama to v1.41.2

Note: the module path has changed to github.com/IBM/sarama since ownership transitioned away from Shopify

Additionally use `sarama.DefaultVersion` as the default config (rather than `sarama.MaxVersion`) if the user hasn't specified an explicit sarama.config.Version in their config-kafka ConfigMap

**Release Note**

```release-note
Migrate to github.com/IBM/sarama and use the latest version for improved protocol support and adherence.
If your target Kafka cluster version is older than 2.1.0 then ensure you set a matching sarama.config.Version in your config-kafka ConfigMap.
```

### CODE REVIEW

**Comment by pierDipi** [comment]
Thank you

**Comment by dnwe** [comment]
In the other tests, strimzi seems to be struggling:

> Error from server (InternalError): error when retrieving current configuration of:
> Resource: "rbac.authorization.k8s.io/v1, Resource=rolebindings", GroupVersionKind: "rbac.authorization.k8s.io/v1, Kind=RoleBinding"
> Name: "strimzi-cluster-operator", Namespace: "kafka"
> from server for: "STDIN": an error on the server ("Internal Server Error: \"/apis/rbac.authorization.k8s.io/v1/namespaces/kafka/rolebindings/strimzi-cluster-operator\": the server is currently unable to handle the request") has prevented the request from succeeding (get rolebindings.rbac.authorization.k8s.io strimzi-cluster-operator)

**Comment by Cali0707** [comment]
@dnwe could you run `hack/update-codegen.sh`? That should fix the build tests

File: control-plane/pkg/kafka/client.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type NewClientFunc func(addrs []string, config *sarama.Config) (sarama.Client, e
 // GetSaramaConfig returns Kafka Client configuration with the given options applied.
 func GetSaramaConfig(configOptions ...ConfigOption) (*sarama.Config, error) {
 	config := sarama.NewConfig()
-	config.Version = sarama.MaxVersion
+	config.Version = sarama.DefaultVersion
```
Comment (matzew) [inline control-plane/pkg/kafka/client.go:59]: Is this due to newer version?

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/retest

**Comment by matzew** [comment]
@dnwe thanks for the help!

**Pull request merged**